itou/prescribers/models.py

```diff
@@ -205,6 +206,15 @@ def must_validate_prescriber_organization_email(self):
         body = "prescribers/email/must_validate_prescriber_organization_email_body.txt"
         return get_email_message(to, context, subject, body)
 
+    @classmethod
+    def get_current_org_or_404(cls, request, return_none_if_not_set=False):
+        pk = request.session.get(settings.ITOU_SESSION_CURRENT_PRESCRIBER_ORG_KEY)
+        if return_none_if_not_set and not pk:
+            return None
+        queryset = cls.objects.member_required(request.user)
```

Comment:
Ici c'est bizarre. On nomme la fonction `get_current_org_or_404` mais en fait plutôt que de retourner une organisation ou une 404, on peut aussi retourner `None`.

Le paramètre `return_none_if_not_set` me semble difficile à comprendre. Il semble pouvoir introduire une possibilité de bug : si on se trompe et qu'on appelle la fonction en forçant `return_none_if_not_set=False` ou en oubliant de le mettre à `True` et que `pk` vaut `None` on retournera une 404 dans tous les cas même pour un prescripteur qui n'a pas d'organisation.

Du coup je me demande si on ne part pas dans un refactoring un peu bancal et s'il ne faudrait pas découpler `get_current_org` et `validate_current_org` le cas échéant au prix d'un peu plus de verbosité.

Reply:
DNRY 
![image](https://user-images.githubusercontent.com/10533583/82428853-c4bfde00-9a93-11ea-8a9d-2ee5bb31f9df.png)

https://itou-inclusion.slack.com/archives/CQ6D0QPPZ/p1589964469004200?thread_ts=1589960167.000400&cid=CQ6D0QPPZ